src/Set.ts

```diff
@@ -0,0 +1,118 @@
+import global from './global';
+import Symbol from './Symbol';
+import { IterableIterator, Iterable, ShimIterator } from './iterator';
+import { hasClass } from './decorators';
+import { forOf } from './iterator';
+
+export namespace Shim {
+	export class Set<T> {
+		private _setData: T[] = [];
+
+		constructor(iterable?: Iterable<T> | ArrayLike<T>) {
+			if (iterable) {
+				forOf(iterable, (value) => this.add(value));
+			}
+		};
+
+		add(value: T): this {
+			if (this.has(value)) {
+				return this;
+			}
+			this._setData.push(value);
+			return this;
+		};
+
+		clear(): void {
+			this._setData.length = 0;
+		};
+
+		delete(value: T): boolean {
+			const idx = this._setData.indexOf(value);
+			if (idx === -1) {
+				return false;
+			}
+			this._setData.splice(idx, 1);
+			return true;
+		};
+
+		entries(): IterableIterator<[T, T]> {
+			const ent = new ShimIterator(this._setData.map((value) => [ value, value ]));
+			(<any> ent)[Symbol.iterator] = function (): IterableIterator<[T, T]> {
```

Comment:
As far as I can tell, none of Dojo 2's target environments support just `Symbol` without `Symbol.iterator`, so this might work in all target environments without a hitch.